stock_storage_type/models/stock_location_storage_type.py

```diff
@@ -114,3 +118,61 @@ def _compute_has_restrictions(self):
                     slst.max_weight,
                 ]
             )
+
+    def _domain_location_storage_type(self, candidate_locations, quants, products):
+        """compute a domain which applies the constraint of the
+        stock.location.storage.type to select locations among candidate
+        locations.
+        """
+        self.ensure_one()
+        location_domain = [
+            ('id', 'in', candidate_locations.ids),
+            ('allowed_location_storage_type_ids', '=', self.id),
+        ]
+        if self.only_empty:
+            location_domain.append(
+                ('location_is_empty', '=', True)
+            )
+        if self.do_not_mix_products:
+            location_domain.append(
+                ('location_will_contain_product_ids', 'not in', products.ids)
```

Comment:
This condition prevent to put a good in a location if it contains the same product, which is not what we want,
we want to exclude a location if it contains other products. (same thing for the lot below)